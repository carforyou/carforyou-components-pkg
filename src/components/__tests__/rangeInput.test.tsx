import React from "react"
import { render } from "@testing-library/react"

import RangeInput from "../rangeInput"

const renderWrapper = ({
  label = null,
  name = "name",
  minValue = null,
  maxValue = null,
  minPlaceholder = "min",
  maxPlaceholder = "max",
  handleChange = null,
} = {}) =>
  render(
    <RangeInput
      label={label}
      name={name}
      value={{
        min: minValue,
        max: maxValue,
      }}
      placeholder={{
        min: minPlaceholder,
        max: maxPlaceholder,
      }}
      handleChange={handleChange}
    />
  )

describe("RangeInput", () => {
  it("renders RangeInput correctly", () => {
    const { container } = renderWrapper()
    expect(container).toMatchSnapshot()
  })

  it("renders label if defined", () => {
    const { getByText } = renderWrapper({ label: "Label" })
    expect(getByText("Label"))
  })
})
