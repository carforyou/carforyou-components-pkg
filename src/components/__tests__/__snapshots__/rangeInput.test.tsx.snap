// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput renders RangeInput correctly 1`] = `
<div>
  <div
    class="flex flex-wrap"
  >
    <div
      class="w-6/12"
    >
      <div
        class="w-12/12 focus-within:text-teal"
      >
        <div
          class="relative pb-px"
        >
          <div
            class="validation-inputContainer mb-20"
          >
            <div
              class="relative flex w-12/12 clearButton-container"
            >
              <input
                class="w-12/12 input_withClearButton input_left hover:z-1 hover:transition hover:duration-200 focus:z-1"
                data-testid="nameFrom"
                data-valid="true"
                id="nameFrom"
                inputmode="numeric"
                name="nameFrom"
                placeholder="min"
                type="number"
                value=""
              />
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
    <div
      class="w-6/12"
    >
      <div
        class="w-12/12 focus-within:text-teal"
      >
        <div
          class="relative pb-px"
        >
          <div
            class="validation-inputContainer mb-20"
          >
            <div
              class="relative flex w-12/12 clearButton-container"
            >
              <input
                class="w-12/12 input_withClearButton input_right -ml-1 hover:z-1 hover:transition hover:duration-200 focus:z-1"
                data-testid="nameTo"
                data-valid="true"
                id="nameTo"
                inputmode="numeric"
                name="nameTo"
                placeholder="max"
                type="number"
                value=""
              />
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
